Wazuh Cloud - Cold Storage Documentation (#7223)
wazuh/wazuh-documentation: source/cloud-service/archive-data/access.rst

      See the :doc:`Wazuh Cloud CLI </cloud-service/cli/index>` section to learn how to list and download your archive data automatically.

> Generate your API key
> ---------------------
> 
> To obtain the api key token it is necessary to execute this step described in the `Wazuh Cloud CLI </cloud-service/cli/index>` section.

Reviewer: ```suggestion
Getting your API key and the AWS token
--------------------------------------

#. Obtain your Wazuh Cloud API key by following the steps outlined in the API :doc:`Authentication </cloud-service/apis/authentication>` section.
```

> Get the AWS token
> ------------------
> 

Reviewer: ```suggestion
```

> Use the ``POST /storage/token`` endpoint of the :cloud-api-ref:`Wazuh Cloud API <tag/storage>` to get the AWS token and access the archive data of a specific environment. In this example, we generate an AWS token valid for 3600 seconds for environment `012345678ab`.

Reviewer: ```suggestion
#. Use the :cloud-api-ref:`POST /storage/token <tag/storage>` API endpoint with your key to get a temporary AWS token. For example, the following request generates an AWS token valid for ``3600`` seconds that grants access to the environment archive data with ID ``012345678ab``.
```

   .. code-block::

      curl -XPOST https://api.cloud.wazuh.com/v2/storage/token -H "x-api-key: <YOUR_API_KEY>" -H "Content-Type: application/json" --data '
      {
         "environment_cloud_id": "012345678ab",
         "token_expiration": "3600"
      }'

   .. code-block:: console
      :class: output

> 

Reviewer: Please highlight the relevant information lines with `:emphasize-lines:`, for example

```
   .. code-block:: console
      :class: output
      :emphasize-lines: 7-9

      {
         "environment_cloud_id": "012345678ab",
         "aws": {
            "s3_path": "wazuh-cloud-cold-us-east-1/012345678ab",
            "region": "us-east-1",
            "credentials": {
               "access_key_id": "mUdT2dBjlHd...Gh7Ni1yZKR5If",
               "secret_access_key": "qEzCk63a224...5aB+e4fC1BR0G",
               "session_token": "MRg3t7HIuoA...4o4BXSAcPfUD8",
               "expires_in": 3600
            }
         }
      }
```

         }
      }

> Generate AWS `wazuh_cloud_storage` profile
> ----------------------------------------

Reviewer: ```suggestion
Generating the AWS `wazuh_cloud_storage` profile
------------------------------------------------
```

Add the token to the AWS credentials file ``~/.aws/credentials``.

   .. code-block:: console

>       

Reviewer: Highlight the relevant lines with `:emphasize-lines:`.

      aws_secret_access_key = qEzCk63a224...5aB+e4fC1BR0G
      aws_session_token = MRg3t7HIuoA...4o4BXSAcPfUD8

> 

Reviewer: ```suggestion
```

Listing archive data
---------------------

This command lists the archive data files of the environment `012345678ab`.

.. code-block:: console

>   # aws --profile wazuh_cloud_storage --region us-east-1 s3 ls --recursive s3://wazuh-cloud-cold-us-east-1/012345678ab/

Reviewer: ```suggestion
   # aws --profile wazuh_cloud_storage --region us-east-1 s3 ls --recursive s3://wazuh-cloud-cold-us-east-1/012345678ab/
```

.. code-block:: none

>   :class: output
> 
>   2024-04-19 17:50:06        493 012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz
>   2024-04-19 18:00:05      77759 012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2100_kdBY42OvE9QJuiia.json.gz

Reviewer: ```suggestion
   :class: output

   2024-04-19 17:50:06        493 012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz
   2024-04-19 18:00:05      77759 012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2100_kdBY42OvE9QJuiia.json.gz
```

Examples
--------

> Downloading archive data (multiple files)
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
Downloading archive data – Multiple files
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

> This command downloads in the `/home/test` directory the archive data files of the environment `012345678ab`.

Reviewer: ```suggestion
This command downloads the archive data files of the environment ``012345678ab`` into the ``/home/test/`` directory.
```

.. code-block:: console

>   # aws --profile wazuh_cloud_storage --region us-east-1 s3 cp --recursive s3://wazuh-cloud-cold-us-east-1/012345678ab/ /home/test/

Reviewer: ```suggestion
   # aws --profile wazuh_cloud_storage --region us-east-1 s3 cp --recursive s3://wazuh-cloud-cold-us-east-1/012345678ab/ /home/test/
```

.. code-block:: none
  :class: output

  download: s3://wazuh-cloud-cold-us-east-1/012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz to output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz

>   download: s3://wazuh-cloud-cold-us-east-1/012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2100_kdBY42OvE9QJuiia.json.gz to output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2100_kdBY42OvE9QJuiia.json.gz

Reviewer: Use three-spaces tabulation

> Downloading archive data (single file)
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
Downloading archive data – Single file
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

> This command downloads in the directory `/home/test` the file `012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz` from the environment `012345678ab`.

Reviewer: ```suggestion
This command downloads the ``012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz`` file of the environment ``012345678ab`` into the directory ``/home/test``.
```

.. code-block:: console

>   # aws --profile wazuh_cloud_storage --region us-east-1 s3 cp --recursive s3://wazuh-cloud-cold-us-east-1/012345678ab/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz /home/test/

Reviewer: ```suggestion
   # aws --profile wazuh_cloud_storage --region us-east-1 s3 cp --recursive s3://wazuh-cloud-cold-us-east-1/012345678ab/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz /home/test/
```

.. code-block:: none

>   :class: output

Reviewer: ```suggestion
   :class: output
```

>   download: s3://wazuh-cloud-cold-us-east-1/012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz to ./012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz

Reviewer: ```suggestion
   download: s3://wazuh-cloud-cold-us-east-1/012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz to ./012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz
```
